server-spi/src/main/java/org/keycloak/models/IdentityProviderModel.java

```diff
@@ -64,6 +66,8 @@
 
     private String displayName;
 
+    private IdentityProviderSyncMode syncMode;
```

Comment:
+1

Reply:
I removed the field and relied on the config (the getter will then provide the default value "LEGACY" in case no value exists). With this change I removed all changes to the representation end entity and there converters.